frontend/src/components/pages/UserManagement/index.tsx

```diff
@@ -15,68 +19,206 @@ import {
+  const buildTableUsers = (): UserMgmtTableRecord[] => {
+    const mergedUsers: UserMgmtTableRecord[] = [];
+
+    volunteers.forEach(function (volunteer) {
+      mergedUsers.push({
+        userId: volunteer.userId,
+        id: volunteer.id,
+        pointOfContact: `${volunteer.firstName} ${volunteer.lastName}`,
+        company: "",
+        email: volunteer.email,
+        phoneNumber: volunteer.phoneNumber,
+        accountType: volunteer.role,
+        approvalStatus: volunteer.status,
+      });
+    });
 
+    donors.forEach(function (donor) {
+      mergedUsers.push({
+        userId: donor.userId,
+        id: donor.id,
+        pointOfContact: `${donor.firstName} ${donor.lastName}`,
+        company: donor.businessName,
+        email: donor.email,
+        phoneNumber: donor.phoneNumber,
+        accountType: donor.role,
+        approvalStatus: "",
+      });
+    });
+
+    return mergedUsers;
+  };
+
+  // Update users whenever donors or volunteers have a change
   React.useEffect(() => {
     const getDonors = async () => {
       const res = await DonorAPIClient.getAllDonors();
       setDonors(res);
     };
 
+    const getVolunteers = async () => {
+      const res = await VolunteerAPIClient.getAllVolunteers();
+      setVolunteers(res);
+    };
+
     getDonors();
+    getVolunteers();
   }, []);
 
+  React.useEffect(() => {
+    setUsers(buildTableUsers());
+  }, [donors, volunteers]);
+
+  const tableData = React.useMemo(() => {
+    let filteredUsers;
+    if (selectedFilter === AccountFilterType.ALL) filteredUsers = users;
+    else if (selectedFilter === AccountFilterType.VOLUNTEER)
+      filteredUsers = users.filter(
+        (user) => user.accountType === Role.VOLUNTEER,
+      );
+    // selectedFilter === AccountFilterType.DONOR
+    else
+      filteredUsers = users.filter((user) => user.accountType === Role.DONOR);
+
+    if (!search) return filteredUsers;
+    return filteredUsers.filter(
+      (user) =>
+        user.accountType.toLowerCase().includes(search.toLowerCase()) ||
+        user.pointOfContact.toLowerCase().includes(search.toLowerCase()) ||
+        user.company.toLowerCase().includes(search.toLowerCase()) ||
+        user.email.toLowerCase().includes(search.toLowerCase()) ||
+        user.phoneNumber.toLowerCase().includes(search.toLowerCase()),
+    );
+  }, [search, selectedFilter, users]);
+
+  // Sets volunteer status to Approved
+  const handleApprove = async (user: UserMgmtTableRecord) => {
+    console.log("ON APPROVE CLICK");
+    const newVolunteerData = { status: Status.APPROVED };
+    const updatedVolunteerResponse: VolunteerDTO = await VolunteerAPIClient.updateVolunteerByUserId(
+      user.userId,
+      newVolunteerData,
+    );
+    console.log("AFTER BE CALL");
+    console.log("updatedResponse", updatedVolunteerResponse);
+    if (updatedVolunteerResponse) {
+      const i = users.findIndex((u) => u.id === user.userId);
```

Comment:
04.01 - having an issue with approving volunteers. We want the 'Approve' button to be gone once an admin has clicked it and changed the selected volunteer's approval status. On clicking, we do hit `updateVolunteerByUserId` and it correctly updates the database. But it doesn't return a response (returns nothing), which is causing problems here since `updatedVolunteerResponse` is not there. But we updated `VolunteerAPIClient` and `VolunteerService.updateVolunteerByUserId` to return stuff

Reply:
the new commit should fix this issue